dags/utils/cma_utils.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+from datetime import datetime
+from importlib import import_module
+from pathlib import Path
+import requests
+import pandas as pd
+from airflow.operators.python import PythonOperator
+from airflow import DAG
+import re
+
+logger = logging.getLogger(__name__)
+
+env = Path(__file__).parent.parent.name
+
+utils = import_module(f"{env}.utils.utils")
+api_utils = import_module(f"{env}.utils.api_utils")
+mapping_utils = import_module(f"{env}.utils.mapping_utils")
+
+
+def get_data_from_artisanat_api(limit=100, offset=100):
+    url = "https://data.artisanat.fr/api/explore/v2.1/catalog/datasets/reparacteurs/records"
+    params = {
+        "limit": limit,
+        "offset": offset,
+    }
+
+    response = requests.get(url, params=params)
+    if response.status_code == 200:
+        data = response.json()
+        return data
+    else:
+        response.raise_for_status()
+
+
+def fetch_data_task(dag: DAG) -> PythonOperator:
+    return PythonOperator(
+        task_id="fetch_data_from_api",
+        python_callable=get_cma_dataset,
+        dag=dag,
+    )
+
+
+def clean_float(value):
+    if isinstance(value, str):
+        value = re.sub(r",$", "", value).replace(",", ".")
+    try:
+        return float(value)
+    except ValueError:
+        return None
+
+
+def create_actors_task(dag: DAG) -> PythonOperator:
+    return PythonOperator(
+        task_id="create_actors",
+        python_callable=create_actors,
+        dag=dag,
+    )
+
+
+def combine_categories(row):
+    categories = [row["categorie"], row["categorie2"], row["categorie3"]]
+    categories = [cat for cat in categories if pd.notna(cat)]
+    return " | ".join(categories)
+
+
+def get_cma_dataset():
+    total_records = get_data_from_artisanat_api(limit=1, offset=0)["total_count"]
+    records_per_request = 100
+    all_data = []
+
+    for offset in range(0, total_records, records_per_request):
+        data = get_data_from_artisanat_api(limit=records_per_request, offset=offset)
+        all_data.extend(data["results"])
+    df = pd.DataFrame(all_data)
+    df["produitsdechets_acceptes"] = df.apply(combine_categories, axis=1)
+
+    return df
+
+
+def create_actors(**kwargs):
```

Comment:
If we can't use the same function than for create_actors from EO, I would prefer to rename this one (avoid confusion)

However, I would prefer to use the same than for EO